azure_pipelines/pr-validation.yml

```diff
@@ -84,6 +84,55 @@ jobs:
       failTaskOnFailedTests: true
       testRunTitle: 'Test Run - $(target)'
 
+- job: fetch_pr_labels
+  displayName: 'Check for PR Label'
+  timeoutInMinutes: 5
+  pool:
+    vmImage: 'macOS-14'
+  steps:
+    - script: |
+        url="https://api.github.com/repos/$BUILD_REPOSITORY_ID/issues/$SYSTEM_PULLREQUEST_PULLREQUESTNUMBER/labels"
```

Comment:
If you wish, you can use github3 library to interact with github APIs like here https://github.com/AzureAD/microsoft-authentication-library-for-objc/blob/021181d8d48b31e00acb6c364de9d94367f4a7d4/azure_pipelines/automation.yml#L55

Reply:
Thank you Ameya, this is good advice. We wanted to do an authenticated request in first place, but before that, I had to do an investigation on how to get the token from Github, and given the time constraints, we decided to move on with the current approach and leave the investigation for the next quarter.

So I'm adding this info about the `github3` library to the PBI we have for the next quarter. Thanks.